src/ucp/wireup/address.c

```diff
@@ -655,8 +652,9 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
     ptr               = UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
 
     if (pack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
-        *(uint64_t*)ptr = worker->uuid;
-        ptr             = UCS_PTR_TYPE_OFFSET(ptr, worker->uuid);
+        *(uint64_t*)ptr    = worker->uuid;
+        *address_header_p |= UCP_ADDRESS_HEADER_FLAG_UUID;
```

Comment:
1. does it affect wire compatibility?
2. how is it possible an address would not have uuid?

Reply:
1. No, this bit was not used/checked before and we still have some bits to define an extension
2. For instance we do not pack it for CM flow (e.g. in `ucp_cm_ep_client_initial_config_get`). Even though it is not supposed to be visible for the user, there is a possibility (and interface) to not pack it. Would need to protect from such cases if they arise in the future.

Reply:
i think we better not define such bit and somehow assert that worker address always has guid